### PR TITLE
EOS-18823: Mini Provisioner: Restructure s3setup_prereqs.json by separating prerequistes sections for each phase

### PR DESCRIPTION
**Pull request by pratyush-seagate**
commit 206feafa23c23fc0cc65c860c1c4f577b1e9af9c
Author: pratyush-seagate <pratyush.k.khan@seagate.com>
Date:   Mon Mar 22 03:23:33 2021 -0600

        EOS-18823: Mini Provisioner: Restructure s3setup_prereqs.json by separating prerequistes sections for each phase
        Change description:
                Restructure s3setup_prereqs.json for each phase as described in the ticket's DoD and assocaited code changes in the respective phases and in setupc
            Added/modified/deleted files:
                modified:   scripts/install_prereqs.sh
                modified:   scripts/provisioning/_s3_setup
                modified:   scripts/provisioning/cleanupcmd.py
                modified:   scripts/provisioning/configcmd.py
                modified:   scripts/provisioning/initcmd.py
                modified:   scripts/provisioning/postinstallcmd.py
                modified:   scripts/provisioning/preparecmd.py
                modified:   scripts/provisioning/resetcmd.py
                modified:   scripts/provisioning/s3setup_prereqs.json
                modified:   scripts/provisioning/setupcmd.py
                modified:   scripts/provisioning/testcmd.py
            Unit test:
                Full single node mini-provisioner deployment using install_prereqs.sh on local VM with various edits in s3setup_prereqs.json to test corner cases

    Signed-off-by: pratyush-seagate <pratyush.k.khan@seagate.com>

### CODE REVIEW

File: scripts/provisioning/s3setup_prereqs.json [hunk 1]
```diff
@@ -1,4 +1,6 @@
 {
+"post_install" :
+  {
     "rpms": ["openldap-servers", "openldap-clients", "haproxy", "motr", "python36-ldap"],
```
Comment (nileshgovande) [inline scripts/provisioning/s3setup_prereqs.json:4]: motr - > cortx-motr

File: scripts/provisioning/s3setup_prereqs.json [hunk 2]
```diff
@@ -1,4 +1,6 @@
...
     "pip3s": ["s3confstore", "cryptography", "s3haproxyconfig", "s3cipher"],
     "services": ["sshd", "rabbitmq-server", "haproxy", "slapd", "rsyslog"],
```
Comment (nileshgovande) [inline scripts/provisioning/s3setup_prereqs.json:6]: Remove services from post-instal

File: scripts/provisioning/s3setup_prereqs.json [hunk 3]
```diff
@@ -10,4 +12,38 @@
 		"file:/opt/seagate/cortx/s3/install/haproxy/logrotate/haproxy",
 		"file:/opt/seagate/cortx/s3/install/haproxy/rsyslog.d/haproxy.conf",
 		"file:/opt/seagate/cortx/s3/install/haproxy/logrotate/logrotate"]
+  },
+"config" :
+  {
+    "rpms": ["openldap-servers", "openldap-clients", "haproxy", "motr", "python36-ldap"],
+    "pip3s": ["s3confstore", "cryptography", "s3haproxyconfig", "s3cipher"],
+    "services": ["sshd", "rabbitmq-server", "haproxy", "slapd", "rsyslog"],
+    "exists": [
```
Comment (nileshgovande) [inline scripts/provisioning/s3setup_prereqs.json:21]: files

File: scripts/provisioning/s3setup_prereqs.json [hunk 4]
```diff
@@ -10,4 +12,38 @@
...
+		"file:/etc/machine-id",
+		"file:/etc/haproxy/haproxy.cfg",
+		"file:/etc/ssl/stx/stx.pem",
+		"file:/opt/seagate/cortx/s3/install/haproxy/503.http",
+		"file:/opt/seagate/cortx/s3/install/haproxy/logrotate/haproxy",
+		"file:/opt/seagate/cortx/s3/install/haproxy/rsyslog.d/haproxy.conf",
+		"file:/opt/seagate/cortx/s3/install/haproxy/logrotate/logrotate"]
+  },
+"init" :
```
Comment (nileshgovande) [inline scripts/provisioning/s3setup_prereqs.json:30]: copy all stuff from config to init

File: scripts/provisioning/s3setup_prereqs.json [hunk 5]
```diff
@@ -10,4 +12,38 @@
...
+  {
+  },
+"reset" :
```
Comment (nileshgovande) [inline scripts/provisioning/s3setup_prereqs.json:33]: RPMs should be everywhere

**Review by nileshgovande — APPROVED** [review]

**Comment by pratyush-seagate** [comment]
@nileshgovande  addressed all the review comments.

**Comment by pratyush-seagate** [comment]
VM deployment seems failed ue to motr's setup failure:


19:08:18    
19:08:18      Traceback (most recent call last):
19:08:18        File \"/opt/seagate/cortx/motr/bin/motr_mini_prov.py\", line 79, in get_current_node
19:08:18          current_node = Conf.get(self._index, 'cluster>server_nodes')[machine_id]
19:08:18      TypeError: 'NoneType' object is not subscriptable
19:08:18    
19:08:18      During handling of the above exception, another exception occurred:
19:08:18    
19:08:18      Traceback (most recent call last):
19:08:18        File \"/opt/seagate/cortx/motr/bin/motr_setup\", line 188, in main
19:08:18          command = Cmd.get_command(desc, argv[1:])
19:08:18        File \"/opt/seagate/cortx/motr/bin/motr_setup\", line 72, in get_command
19:08:18          return args.command(args)
19:08:18        File \"/opt/seagate/cortx/motr/bin/motr_setup\", line 91, in __init__
19:08:18          super().__init__(args)
19:08:18        File \"/opt/seagate/cortx/motr/bin/motr_setup\", line 39, in __init__
19:08:18          self._server_id = get_current_node(self)
19:08:18        File \"/opt/seagate/cortx/motr/bin/motr_mini_prov.py\", line 81, in get_current_node
19:08:18          raise MotrError(errno.EINVAL, \"Current node not found\")
19:08:18      motr_mini_prov.MotrError: error[22]: Current node not found

**Review by vimalraghwani20 — APPROVED** [review]

File: scripts/provisioning/s3setup_prereqs.json [hunk 6]
```diff
@@ -1,13 +1,65 @@
...
+"prepare" :
+  {
+    "rpms": ["openldap-servers", "openldap-clients", "haproxy", "cortx-motr", "python36-ldap"],
+    "pip3s": ["s3confstore", "cryptography", "s3haproxyconfig", "s3cipher"]
+  },
+"cleanup" :
+  {
+    "rpms": ["openldap-servers", "openldap-clients", "haproxy", "cortx-motr", "python36-ldap"],
+    "pip3s": ["s3confstore", "cryptography", "s3haproxyconfig", "s3cipher"],
+    "services": ["haproxy", "slapd"]
```
Comment (nileshgovande) [inline scripts/provisioning/s3setup_prereqs.json:57]: Remove services from cleanup and reset
